### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by AntonioMLancuentra**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
As per assignment 1
## What did you learn from the changes you have made?
How does a neural network works through the custom classes
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I experimented a lot. Still I need to improve the reproducibility, I get it within the session but not for you to reproduce them.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Not challenges, just long because calculations take a bit. Oh wait, I tried to shuffle the dataset before each epoch and it works. But I tried it to be reproducible and it didn't work for me, finally I removed the shuffling.
## How were these changes tested?
Not much testing, just looked a bit to all numbers to see that they made sense.
## A reference to a related issue in your repository (if applicable)
Reproducibility, how to create a seed that does not change when restarting the kernel
## Checklist
- [V] I can confirm that my changes are working as intended

### CODE REVIEW

File: 02_activities/assignments/assignment_1.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_1.ipynb:1]: **Question 2**: You implemented a three-layer neural network model without nonlinearity. Mathematically speaking, by the chain rule of matrix multiplication, this is equivalent to a single layer model with `Dense(10)`.
**Question 3**: You implemented a convolutional neural network with `BatchNorm` and multiple `Dense` layers. Actually, this assignment expects you to implement a simple CNN here and experiment with more parameters in later parts.

**Review by tianyi21 — APPROVED** [review]
Hi Antonio,

Thank you for submitting your DL week 1 assessment.

Please find my comments attached to each file. Your A1, Lab1, and Lab2 are now complete. 

Thanks,
Tianyi [LS]

File: 02_activities/assignments/assignment_1.ipynb
Comment (AntonioMLancuentra) [reply 02_activities/assignments/assignment_1.ipynb:1]: Question 2: I just draw an easy example: input (x1, x2), 1 layer of 3 neurons + 1 layer of 2 neurons. I wrote the equations and I saw what you are saying, we never get x1*x2 or x1*x1, it is always the same linear Ax1+Bx2+C. Thanks for the observation, I didn't see that coming, very nice.

Question 3: true, I just wanted to explore and did not realize that later asked for it. It was already done, even if I realized I left it,  sorry about that.
